### DISCUSSION TITLE
Anisotropic structure in 2D simulation

### DISCUSSION THREAD

**Discussion by ShuangXia111** [facebookresearch/fmmax#118] [Q&A]
To the contributors of this code，
 
Appreciate your valuable code for RCWA calculation, especially the exploration in anisotropic materials！

I noticed that there is an example showing how to calculate 1D anisotropic grating. However, the example only considers the condition that the off-diagonal elements of the dielectric constant tensor appear in xy and yx directions. When I try to calculate a **two-dimensional** periodic structure, such as periodic hole, the off-diagonal elements may appear in **xz and zx directions**. So, how can I implement the calculation of 2D periodic structures composed with anisotropic materials, especially when the dielectric constant tensor has xz or yz components? I wonder if this will require many modifications, such as the algorithms used in the Transfer Matrix Method (TMM) and the scattering matrix calculations.

This issue has been bothering me for some time, and I am eagerly looking forward to your assistance.

Thank you so much!

**Comment by mfschubert** [comment]
Hi @ShuangXia111 , indeed fmmax does not support arbitrary anisotropy, although it would be a nice capability. Although I haven't searched extensively, I am not aware of any references that derive the necessary equations.

Actually, it seems that structures with general anisotropy may not support modes with profile that is independent of z, as the polarization will rotate. Therefore, the fundamental approach of the basic FMM (dealing with such z-invariant modes) may be incompatible with general anisotropy. However, this requires some more investigation.

If you have some references or ideas on how to handle general anisotropy those would be welcome.
